### PR TITLE
Vue3: Add vite plugin for portable stories

### PR DESCRIPTION
**Pull request by yannbf**
Closes #

## What I did

This PR introduces a plugin in `@storybook/vue3-vite` so that users can get the necessary config to test their stories using portable stories:

```ts
import { defineWorkspace, defaultExclude } from "vitest/config";
import { storybookTest } from "@storybook/experimental-addon-vitest/plugin";
import { storybookVuePlugin } from '@storybook/vue3-vite/vite'

export default defineWorkspace([
  {
    extends: "vite.config.ts",
    plugins: [
      storybookTest(),
     storybookVuePlugin() // 👈 Add this plugin, it will allow to mount template-based Vue components
    ],
  }
])
```

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  76.5 MB | 76.5 MB | 0 B | **1.81** | 0% |
| initSize |  161 MB | 161 MB | 0 B | -0.23 | 0% |
| diffSize |  84.8 MB | 84.8 MB | 0 B | -0.24 | 0% |
| buildSize |  7.48 MB | 7.48 MB | 115 B | 0.83 | 0% |
| buildSbAddonsSize |  1.62 MB | 1.62 MB | 0 B | **1.73** | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  2.31 MB | 2.31 MB | 0 B | 0.82 | 0% |
| buildSbPreviewSize |  352 kB | 352 kB | 0 B | - | 0% |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  4.47 MB | 4.47 MB | 0 B | 0.82 | 0% |
| buildPreviewSize |  3.01 MB | 3.01 MB | 115 B | 0.84 | 0% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  23.2s | 6.3s | -16s -807ms | **-1.28** | 🔰-262.7% |
| generateTime |  19.5s | 18.2s | -1s -389ms | -0.9 | -7.6% |
| initTime |  16.8s | 15s | -1s -839ms | -0.77 | -12.3% |
| buildTime |  14.2s | 13.5s | -721ms | 0.7 | -5.3% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  7s | 7.1s | 116ms | -0.22 | 1.6% |
| devManagerResponsive |  4.3s | 4.6s | 276ms | -0.1 | 5.9% |
| devManagerHeaderVisible |  685ms | 791ms | 106ms | -0.23 | 13.4% |
| devManagerIndexVisible |  726ms | 837ms | 111ms | -0.18 | 13.3% |
| devStoryVisibleUncached |  1.1s | 1.4s | 274ms | -0.17 | 19.4% |
| devStoryVisible |  724ms | 837ms | 113ms | -0.18 | 13.5% |
| devAutodocsVisible |  684ms | 678ms | -6ms | -0.52 | -0.9% |
| devMDXVisible |  889ms | 778ms | -111ms | 0.2 | -14.3% |
| buildManagerHeaderVisible |  713ms | 752ms | 39ms | -0.03 | 5.2% |
| buildManagerIndexVisible |  720ms | 759ms | 39ms | -0.06 | 5.1% |
| buildStoryVisible |  795ms | 799ms | 4ms | -0.24 | 0.5% |
| buildAutodocsVisible |  654ms | 735ms | 81ms | 0.58 | 11% |
| buildMDXVisible |  610ms | 670ms | 60ms | -0.19 | 9% |

## Greptile Summary

This PR introduces a new Vite plugin for Vue3 in Storybook, enabling portable story testing and improving Vue3 component support with Vite as a builder.

- Added `storybookVuePlugin` in `code/frameworks/vue3-vite/src/vite.ts` for Vue3 template compilation
- Modified `code/frameworks/vue3-vite/src/preset.ts` to include the new `templateCompilation` plugin
- Updated `code/addons/vitest/src/postinstall.ts` to support Vue3 Vite plugin in Vitest addon
- Enhanced `scripts/utils/main-js.ts` with a more flexible `readConfig` function
- Added `extendPreview` task in `scripts/tasks/sandbox.ts` for additional preview environment configuration

### CODE REVIEW

File: scripts/tasks/sandbox-parts.ts [hunk 1]
```diff
@@ -488,7 +490,6 @@ export async function setupVitest(details: TemplateDetails, options: PassedOptio
               name: "chromium",
               provider: "playwright",
               headless: true,
-              screenshotFailures: false,
```
Comment (shilman) [inline scripts/tasks/sandbox-parts.ts:491]: Is this related?

File: scripts/utils/main-js.ts [hunk 2]
```diff
@@ -4,18 +4,18 @@ import slash from 'slash';
 
 import { getInterpretedFile } from '../../code/core/src/common';
 import type { ConfigFile } from '../../code/core/src/csf-tools';
-import { readConfig } from '../../code/core/src/csf-tools';
+import { readConfig as csfReadConfig } from '../../code/core/src/csf-tools';
 
-export async function readMainConfig({ cwd }: { cwd: string }) {
+export async function readConfig({ fileName = 'main', cwd }: { fileName?: string; cwd: string }) {
```
Comment (shilman) [inline scripts/utils/main-js.ts:9]: I'd suggest making `fileName` required and no default

File: scripts/tasks/sandbox-parts.ts [hunk 3]
```diff
@@ -787,6 +787,17 @@ export const extendMain: Task['run'] = async ({ template, sandboxDir, key }, { d
   await writeConfig(mainConfig);
 };
 
+export const extendPreview: Task['run'] = async ({ template, sandboxDir }) => {
+  logger.log('📝 Extending preview.js');
+  const previewConfig = await readConfig({ cwd: sandboxDir, fileName: 'preview' });
+
+  if (template.expected.builder.includes('vite')) {
+    previewConfig.setFieldValue(['tags'], ['vitest']);
```
Comment (shilman) [inline scripts/tasks/sandbox-parts.ts:795]: I realize this is unrelated to the PR how does the `vitest` tag differ from the `test` tag?

**Review by shilman — APPROVED** [review]
LGTM aside from one minor change

File: scripts/tasks/sandbox-parts.ts (hunk 1 above)
Comment (yannbf) [reply scripts/tasks/sandbox-parts.ts:491]: I'm cleaning up things from changes we made but forgot to clean up. This setting is false by default now

File: scripts/tasks/sandbox-parts.ts (hunk 3 above)
Comment (yannbf) [reply scripts/tasks/sandbox-parts.ts:795]: We use both the test-runner and vitest runner. There are things that we can't test in vitest that we can in the test-runner (e.g. channels). So we decided to disable tests via `!test` for the test-runner and `!vitest` for vitest. Normal users will likely not be in this scenario so `!test` should be enough for them

**Review by JReinhold — APPROVED** [review]
LGTM 🚀
